enterprise_gateway/services/sessions/kernelsessionmanager.py

```diff
@@ -181,7 +188,7 @@ def _pre_save_transformation(sessions):
                 info = session['connection_info']
                 key = info.get('key')
                 if key:
-                    info['key'] = bytes_to_str(key)
+                    info['key'] = unicode_to_str(key)
```

Comment:
The `key` should be handled as bytes.  Is there a reason you found this change to be necessary?

Reply:
Yes, I was getting error saying `AttributeError: 'str' object has no attribute 'decode'` from ~/anaconda2/envs/ana35/lib/python3.5/site-packages/ipython_genutils/py3compat.py:157

Reply:
Anything regarding this error?